src/config.rs

```diff
@@ -9,7 +9,17 @@ use dirs::PROJECT_DIRS;
 use util::transpose;
 
 pub fn config_file() -> PathBuf {
-    PROJECT_DIRS.config_dir().join("config")
+    match env::var("BAT_CONFIG_PATH") {
```

Comment:
An alternative way to write this logic would be with the combinators on `Option`:
```rust
env::var("BAT_CONFIG_PATH")
    .ok()
    .map(PathBuf::from)
    .filter(Path::is_file)
    .unwrap_or(PROJECT_DIRS.config_dir().join("config"))
```

Reply:
FYI I am working on this looks like below, but seems to be causing some weird issues with my build. I'll keep at it :) 

```rust
env::var("BAT_CONFIG_PATH")
        .ok()
        .map(PathBuf::from)
        .filter(|config_path| config_path.is_file())
        .unwrap_or(PROJECT_DIRS.config_dir().join("config"))
```

Reply:
Okay, let me know if you need help.

Reply:
Added two integration tests, one to check if it's reading the environment variable path right, and another that enables paging and checks for the paging output like `pager_basic()`